File: CS/09_Forms/AddCalendarDropdown.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Actions;
using Spire.Pdf.Fields;
using Spire.Pdf.Graphics;


namespace AddCalendarDropdown
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new PDF document
            PdfDocument pdf = new PdfDocument();

            // Add a new page to the PDF document with A4 size and zero margins
            PdfPageBase page = pdf.Pages.Add(PdfPageSize.A4, new PdfMargins(0));

            // Create a TrueType font using Arial Unicode MS with a font size of 10
            PdfTrueTypeFont font = new PdfTrueTypeFont(new Font("Arial Unicode MS", 10f), true);

            // Create a new text box field on the page with the field name "date"
            PdfTextBoxField textbox = new PdfTextBoxField(page, "date");

            // Set the position and size of the text box
            textbox.Bounds = new RectangleF(40, 50, 60, 20);

            // Set the font for the text box
            textbox.Font = font;

            // Get the JavaScript keystroke string for date input in the format "mm/dd/yyyy"
            string kjs = PdfJavaScript.GetDateKeystrokeString("mm/dd/yyyy");

            // Get the JavaScript format string for date input in the format "mm/dd/yyyy"
            string fjs = PdfJavaScript.GetDateFormatString("mm/dd/yyyy");

            // Create JavaScript actions for the keystroke and format settings
            PdfJavaScriptAction kjsAction = new PdfJavaScriptAction(kjs);
            PdfJavaScriptAction fjsAction = new PdfJavaScriptAction(fjs);

            // Assign the JavaScript actions to the text box
            textbox.Actions.KeyPressed = kjsAction;
            textbox.Actions.Format = fjsAction;

            // Add the text box field to the form fields collection of the PDF document
            pdf.Form.Fields.Add(textbox);

            // Save the PDF document to a file
            string result = "AddCalendarDropdown_result.pdf";
            pdf.SaveToFile(result);

            // Close the PDF document
            pdf.Close();

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/09_Forms/AddCalendarDropdown.vb
Imports Spire.Pdf
Imports Spire.Pdf.Actions
Imports Spire.Pdf.Fields
Imports Spire.Pdf.Graphics


Namespace AddCalendarDropdown
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument instance
			Dim pdf As New PdfDocument()

			' Add a new page to the document with A4 size and zero margins
			Dim page As PdfPageBase = pdf.Pages.Add(PdfPageSize.A4, New PdfMargins(0))

			' Create a TrueType font for the text field
			Dim font As New PdfTrueTypeFont(New Font("Arial Unicode MS", 10.0F), True)
			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			'Dim font As New PdfTrueTypeFont("Arial Unicode MS", 10.0F, PdfFontStyle.Regular, True)
			' =============================================================================

			' Create a new text box field on the page named "date"
			Dim textbox As New PdfTextBoxField(page, "date")

			' Set the position and size of the text box
			textbox.Bounds = New RectangleF(40, 50, 60, 20)

			' Set the font for the text box
			textbox.Font = font

			' Get the JavaScript keystroke string for date format (mm/dd/yyyy)
			Dim kjs As String = PdfJavaScript.GetDateKeystrokeString("mm/dd/yyyy")

			' Get the JavaScript format string for date format (mm/dd/yyyy)
			Dim fjs As String = PdfJavaScript.GetDateFormatString("mm/dd/yyyy")

			' Create a JavaScript action for keystroke event
			Dim kjsAction As New PdfJavaScriptAction(kjs)

			' Create a JavaScript action for format event
			Dim fjsAction As New PdfJavaScriptAction(fjs)

			' Assign the JavaScript actions to the text box
			textbox.Actions.KeyPressed = kjsAction
			textbox.Actions.Format = fjsAction

			' Add the text box field to the form
			pdf.Form.Fields.Add(textbox)

			' Specify the output file name
			Dim result As String = "AddCalendarDropdown_result.pdf"

			' Save the modified document to the specified file path
			pdf.SaveToFile(result)

			' Close the document
			pdf.Close()

			' Launch the Pdf file
			PDFDocumentViewer(result)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
